databrowser/src/domain/auth/store/auth.ts

```diff
@@ -35,10 +36,12 @@ const mutations = {
   authenticated(state: State, accessToken: string) {
     state.isAuthenticated = true;
     state.accessToken = accessToken;
+    axios.defaults.headers.common['Authorization'] = `Bearer ${accessToken}`;
```

Comment:
You are right that we are mixing two different things into the mutation method, but we need to keep the state change and axios-header update together, otherwise other developers must know about the details on how to authenticate / unauthentitcate correctly (state change + header update).

What do you think about an auth service / function, that should be called instead of the mutation and that combines the state change and header update? That way we could extract the header change from the mutation and provide a single entry point for authorization.

Reply:
That sounds nice!

Reply:
Instead of implementing a dedicated service, I moved the side-effect part of axios header update, together with the mutation invocation, into vuex actions. Actions are the right place to perform side-effect operations.